lib/session_db.php

```diff
@@ -169,4 +185,94 @@ public function db_start($request) {
             }
         }
     }
+
+    /**
+     * Acquire a lock for the session (unified for all DB types)
+     * @param string $key session key
+     * @return bool true if lock acquired, false otherwise
+     */
+    private function acquire_lock($key) {
+        $lock_name = 'session_lock_' . substr(hash('sha256', $key), 0, 51);        
+        $query = '';
+        $params = [];
+
+        switch ($this->db_driver) {
+            case 'mysql':
+                $query = 'SELECT GET_LOCK(:lock_name, :timeout)';
+                $params = [':lock_name' => $lock_name, ':timeout' => $this->lock_timeout];
```

Comment:
We should implement the lock_timeout in other drivers as well - this is important for concurrent requests. You can use a polling mechanism with intervals of 500ms to 1 second.

Reply:
@kroky I just updated, I just did a test with mysql. I'm waiting to see if there are any corrections from you so far before confirming with `pgsql` too but it should work for `pgsql` too

Reply:
The polling looks good.